tests/test_updater_with_simulator.py

```diff
@@ -254,6 +255,32 @@ def test_not_loading_targets_twice(self, wrapped_open: MagicMock) -> None:
         updater.get_targetinfo("somepath")
         wrapped_open.assert_not_called()
 
+    def test_expired_metadata(self) -> None:
+        # Test that expired local timestamp/snapshot can be used for updating
+        # from remote
+
+        # Make a successful update of valid metadata which stores it in cache
+        self._run_refresh()
+
+        # Simulate expired local metadata by mocking system time one second ahead
+        mock_time = Mock()
+        mock_time.return_value = (
+            int(self.sim.timestamp.expires.strftime("%Y%m%d%H%M%S")) + 1
+        )
+        with patch("time.time", mock_time):
```

Comment:
I am not sure I understand the exact intentions but the tests pass because patch("time.time", mock_time) does not work as you expected and there isn't any expired metadata. The tests would work the exact same way if you delete the lines doing the "mocking". 
Updater (TrustedMetadataSet) uses datetime.utcnow to calculate the reference time.